lib/analytics_creator.rb

```diff
@@ -0,0 +1,66 @@
+class AnalyticsCreator
+  SCHEMA = "public"
+  attr_reader :username, :password, :database_name
+
+
+  def initialize(username:, password:, database_name: nil)
+    @username = username
+    @password = password
+    @database_name = database_name || ActiveRecord::Base.connection_db_config.database
+  end
+
+  def run
+    create_user
+    apply_permissions
+  end
+
+  def drop_user
+    roles = execute("select * from pg_catalog.pg_roles where rolname='#{username}'")
+    return unless roles.rows.any?
+
+    execute("REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA #{SCHEMA} FROM #{username};")
+    execute("REVOKE USAGE ON SCHEMA #{SCHEMA} FROM #{username}")
+    execute("ALTER DEFAULT PRIVILEGES IN SCHEMA #{SCHEMA} REVOKE ALL ON TABLES FROM #{username}")
+    execute("REVOKE CONNECT ON DATABASE #{database_name} FROM #{username}")
+    execute("DROP ROLE #{username}")
+  end
+
+private
+
+  def create_user
+    raise "ANALYTICS_DB_PASSWORD must be set." if password.blank?
+
+    roles = execute("select * from pg_catalog.pg_roles where rolname='#{username}'")
+    return if roles.rows.any?
+
+    execute("CREATE ROLE #{username} WITH LOGIN PASSWORD '#{password}'")
+    execute("GRANT CONNECT ON DATABASE #{database_name} TO #{username}")
+    execute("GRANT USAGE ON SCHEMA #{SCHEMA} TO #{username}")
+  end
+
+  def create_user_non_public_schema(schema)
+    raise "ANALYTICS_DB_PASSWORD must be set." if password.blank?
+    raise "can not be run against public schem" if schema == 'public'
+
+    roles = execute("select * from pg_catalog.pg_roles where rolname='#{username}'")
+    return if roles.rows.any?
+
+    execute("CREATE ROLE #{username} WITH LOGIN PASSWORD '#{password}'")
+    execute("GRANT CONNECT ON DATABASE #{database_name} TO #{username}")
+    execute("CREATE SCHEMA IF NOT EXISTS #{schema}")
+    execute("GRANT USAGE ON SCHEMA #{schema} TO #{username}")
+    execute("grant select on all tables in schema #{schema} to #{username}")
+    execute("alter default privileges in schema #{schema} grant select on tables to #{username}")
+    execute("ALTER ROLE #{username} SET search_path TO #{schema}")
```

Comment:
be good to understand why all of these are necessary

Reply:
it's not, but I'm trying to decide if it is worth leaving in so this would be used if scenic supported multiple DB (then you could:
* create an analytics schema
* give the analytics users full permission to view tables in the schema
* remove the rake job to set access rights

I will most likely remove it...